submodules/k8s/templates/k8s-functions.sh.tftpl

```diff
@@ -34,42 +22,49 @@ check_kubeconfig() {
 }
 
 set_k8s_auth() {
-  AWS_AUTH_YAML="${aws_auth_yaml}"
+  local AWS_AUTH_YAML="${aws_auth_yaml}"
   if test -f "$AWS_AUTH_YAML"; then
     printf "$GREEN Updating $AWS_AUTH_YAML... $EC \n"
-    kubectl apply -f "$AWS_AUTH_YAML"
+    kubectl_apply "$AWS_AUTH_YAML"
   else
     printf "$RED $AWS_AUTH_YAML does not exist. $EC \n" && exit 1
   fi
   echo
 }
 
 install_calico() {
-  CALICO_OPERATOR_YAML_URL=${calico_operator_url}
+  local CALICO_OPERATOR_YAML_URL=${calico_operator_url}
   printf "$GREEN Installing Calico Operator $EC \n"
-  kubectl apply -f $CALICO_OPERATOR_YAML_URL || printf "$RED There was an error installing the calico operator"
+  kubectl_apply $CALICO_OPERATOR_YAML_URL || printf "$RED There was an error installing the calico operator"
   echo
-  CALICO_CRD_YAML_URL=${calico_custom_resources_url}
+  local CALICO_CRD_YAML_URL=${calico_custom_resources_url}
   printf "$GREEN Installing Calico Custom resources $EC \n" || printf "$RED There was an error installing the calico CRD"
-  kubectl apply -f $CALICO_CRD_YAML_URL
+  kubectl_apply $CALICO_CRD_YAML_URL
   echo
 }
 
+validate_url() {
+  local url="$1"
+  if curl --head --silent --fail -o /dev/null "$url"; then
+    return 0
+  else
+    return 1
+  fi
```

Comment:
why not send back the rc ($?) ?

Reply:
We can, since any non-zero will work, Preferred an explicit conditional return

Reply:
I just realized since this is silent, do we need to say something on error.

Reply:
its covered by the client call 
```
  if test -f "$k8s_manifest" || validate_url "$k8s_manifest"; then
    echo "Applying $k8s_manifest..."
    (
      export HTTPS_PROXY=socks5://127.0.0.1:${k8s_tunnel_port} && kubectl apply -f $k8s_manifest
    )
  else
    printf "$RED $k8s_manifest does not exist. $EC \n" && exit 1
  fi
```

Reply:
no, that is the first half, not the validate_url which we have no real idea why it failed.